### PR TITLE
ks.read_delta / DataFrame.to_delta

### LINKED ISSUE #444: Implement read_delta and DataFrame.to_delta
Would be cool to have Delta support built-in so it's easy for users to read/write Delta Lake tables.

Once we implement this, we should update read_csv / write_csv docs to suggest users to use delta or parquet instead of csv.

### PR DESCRIPTION
**Pull request by rxin**
Still need to figure out how to do testing (primarily how to load the dependency in).

Resolves #444

### CODE REVIEW

**Comment by rxin** [comment]
@HyukjinKwon any idea how to inject delta oss dependency into our test environment?

**Comment by HyukjinKwon** [comment]
Let me check

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -1554,6 +1554,53 @@ def to_table(self, name: str, format: Optional[str] = None, mode: str = 'error',
...
+        DataFrame.to_parquet
+        DataFrame.to_table
+        DataFrame.to_spark_io
+
+        Examples
+        --------
+
+        Create a new Delta Lake table, partitioned by one column:
+
+        >>> df.to_delta('/delta/events', partition_cols='date')  # doctest: +SKIP
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: i'm fine to merge as is. I can take a look separately as well.

**Comment by rxin** [comment]
We shouldn’t merge it at least without trying to see if it works....

**Comment by HyukjinKwon** [comment]
K.. I thought you checked it..

**Comment by HyukjinKwon** [comment]
let me take this PR over

**Comment by rxin** [comment]
Closing this and moving to https://github.com/databricks/koalas/pull/456

**Pull request closed without merging**